src/commands/tasks/taskConfig.ts

```diff
@@ -0,0 +1,182 @@
+import * as fs from 'fs/promises';
+import * as ospath from 'path';
+import { limits, runtimeChoices } from '../../constants';
+
+export enum permissionModes {
+  permissionRequired = 'permissionRequired',
+  allUsers = 'allUsers',
+  public = 'public',
+}
+
+export interface TaskConfig {
+  name?: string;
+  path?: string;
+  entryPoint?: string;
+  runtime?: string;
+  description?: string;
+  timeLimit?: number;
+  memoryLimit?: number;
+  environment?: Record<string, any>;
+  executionPermission?: permissionModes;
+}
+
+function replaceConfigVariables(config: any): any {
+  let result: any;
+
+  switch (typeof config) {
+    case 'object':
+      if (!Array.isArray(config)) {
```

Comment:
Swap if-else block to have the positive flow in the condition.
I always read over the '!' in front of conditions which caused confusion here. I try to always write my conditions in a positive sense or extract the condition to a function which shows the intent (e.g. assert func to validate input). `Array.isArray(config)` rather than `!Array.isArray(config)`